builtin/logical/plugin/backend.go

```diff
@@ -0,0 +1,39 @@
+package plugin
```

Comment:
I wonder if this should go under `builtin/plugin/backend.go` instead since this can be used for both logical and credential backends.